### ISSUE TITLE
StringCollectionAssertions.Contain overloads

### ISSUE THREAD

**Issue by Meir017** [dennisdoomen/fluentassertions#219]
When writing something like this:
```csharp
IEnumerable<string> actual = new[] {"text1", "text2", "text3"};
IEnumerable<string> expected = new[] {"text1", "text2"};
actual.Should().Contain(expected, "because some reason");
```
The Overloads:
```csharp
public AndConstraint<StringCollectionAssertions> Contain(IEnumerable<string> expectedItemsList,
            string additionalExpectedItem);
public AndConstraint<StringCollectionAssertions> Contain(IEnumerable<string> expected, string because, object reasonArg, params object[] reasonArgs);
```
The only wasy I was able to use the overload with the because was writing something like this:
```csharp
actual.Should().Contain(expected, "{0}", "because some reason");
```

**Comment by dennisdoomen** [comment]
Thanks for reporting this. Could you please try one the latest pre-release version from https://www.myget.org/gallery/fluentassertions?

**Comment by Meir017** [comment]
Tried vertion - v3.3.0-unstable0139, didn't solve the problem.

Is the overload with the parameter 'additionalExpectedItem' really needed?

**Comment by dennisdoomen** [comment]
What if you do this?

    actual.Should().Contain(expected.ToArray(), "because some reason");

**Comment by Meir017** [comment]
Tried all of the options.
Here is the code I used
```csharp
[TestClass]
    public class StringCollectionContainTests
    {
        [TestMethod]
        public void EnumerableContainNoBecauseArguments()
        {
            IEnumerable<string> actual = new[] { "item1", "item2", "item3", "item4" };
            IEnumerable<string> expected = new[] { "item2", "item3" };
            actual.Should().Contain(expected, "because some reason"); // fails
        }

        [TestMethod]
        public void EnumerableContainWithBecauseArguments()
        {
            IEnumerable<string> actual = new[] { "item1", "item2", "item3", "item4" };
            IEnumerable<string> expected = new[] { "item2", "item3" };
            actual.Should().Contain(expected, "{0}", "because some reason"); // passes
        }

        [TestMethod]
        public void ArrayContainNoBecauseArguments()
        {
            IEnumerable<string> actual = new[] { "item1", "item2", "item3", "item4" };
            IEnumerable<string> expected = new[] { "item2", "item3" };
            actual.Should().Contain(expected.ToArray(), "because some reason"); // fails
        }

        [TestMethod]
        public void ArrayContainWithBecauseArguments()
        {
            IEnumerable<string> actual = new[] { "item1", "item2", "item3", "item4" };
            IEnumerable<string> expected = new[] { "item2", "item3" };
            actual.Should().Contain(expected.ToArray(), "{0}", "because some reason"); // passes
        }
    }
```

**Comment by vossad01** [comment]
It looks like it is not an overload resolution issue, just an unexpected set of overloads in FA.

These methods do not seem to be very idiomatic for Fluent Assertions.  The string parameter would typically be the `because` reason not an additional.  These overloads seem to have a long history though -- at least since v2.1 -- first being introduced for collections in fb7c69707c1982244ea3add36480080d55229c52

Users could get the same functionality by:
`collection.Should().Contain(strArray2).And.Contain(strArray2)` or,
`collection.Should().Contain(strArray.Concat(strArray2))`

Strings may be a special case because the type of collection items is the same as the usual `because` parameter.

My first reaction is to deprecate these overloads marking for removal in 4.0,
```c#
public AndConstraint<StringCollectionAssertions> Contain(
    IEnumerable<string> expectedItemsList,
    string additionalExpectedItem)
public AndConstraint<StringCollectionAssertions> Contain(
    IEnumerable<string> expectedItemsList,
    IEnumerable<string> additionalExpectedItems)
```

As well as add optional parameters to the following overload,
```c#
public AndConstraint<StringCollectionAssertions> Contain(
    IEnumerable<string> expected,
    string because,
    object reasonArg,
    params object[] reasonArgs)
```

This would allow users to get the correct overload by specifying the parameter name until 4.0,
```c$
actual.Should().Contain(expected, because: "some reason");
```

I am surprised by these overloads but given how long they have been part of FA, I am guessing it is appropriate to leave for other types of collections.

**Comment by dennisdoomen** [comment]
Agreed. I actually vaguely remember why they were added. It had something to do with asserting that an extra item was added to a collection. So when you had `collection` and your production code would tuck another item `addedItem` to it, you could use these overloads to verify that behavior. E.g.

  collection.Should().Contain(collection, addedItem);

Without it, you had to do something like this:

  collection.Should().Contain(collection.Concat(new[] {addedItem]));

**Comment by dennisdoomen** [comment]
Fixed by PR #221.
